tests/Builder/Stage/AddFieldsStageTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Tests\Builder\Stage;
+
+use MongoDB\Builder\Expression;
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Stage;
+use MongoDB\Tests\Builder\PipelineTestCase;
+
+class AddFieldsStageTest extends PipelineTestCase
+{
+    /**
+     * THIS METHOD IS AUTO-GENERATED. ANY CHANGES WILL BE LOST!
+     *
+     * @see testAddingFieldsToAnEmbeddedDocument
+     *
+     * @return list<array<string, mixed>>
+     */
+    public function getExpectedAddingFieldsToAnEmbeddedDocument(): array
+    {
+        return [(object) [
+            '$addFields' => (object) ['specs.fuel_type' => 'unleaded'],
+        ],
+        ];
+    }
+
+    /**
+     * THIS METHOD IS AUTO-GENERATED. ANY CHANGES WILL BE LOST!
+     *
+     * @see testUsingTwoAaddFieldsStages
+     *
+     * @return list<array<string, mixed>>
+     */
+    public function getExpectedUsingTwoAaddFieldsStages(): array
+    {
+        return [
+            (object) [
+                '$addFields' => (object) [
+                    'totalHomework' => (object) ['$sum' => '$homework'],
+                    'totalQuiz' => (object) ['$sum' => '$quiz'],
+                ],
+            ],
+            (object) [
+                '$addFields' => (object) [
+                    'totalScore' => (object) [
+                        '$add' => [
+                            '$totalHomework',
+                            '$totalQuiz',
+                            '$extraCredit',
+                        ],
+                    ],
+                ],
+            ],
+        ];
+    }
+
+    /** @see getExpectedAddingFieldsToAnEmbeddedDocument */
+    public function testAddingFieldsToAnEmbeddedDocument(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::addFields(
+                ...['specs.fuel_type' => 'unleaded'],
+            ),
+        );
+
+        $expected = $this->getExpectedAddingFieldsToAnEmbeddedDocument();
+
+        $this->assertSamePipeline($expected, $pipeline);
+    }
+
+    /** @see getExpectedUsingTwoAaddFieldsStages */
+    public function testUsingTwoAaddFieldsStages(): void
+    {
+        $this->markTestSkipped('$sum must accept arrayFieldPath and render it as a single value: https://jira.mongodb.org/browse/PHPLIB-1287');
```

Comment:
This pipeline can't be implemented. Work on `$sum` operator is required. [PHPLIB-1287](https://jira.mongodb.org/browse/PHPLIB-1287)